tests/monotouch-test/ARKit/ARPlaneGeometryTest.cs

```diff
@@ -0,0 +1,145 @@
+
+		GCHandle verticesArrayHandle;
+		GCHandle textureCoordinatesArrayHandle;
+		GCHandle indicesArrayHandle;
+		GCHandle boundaryVerticesArrayHandle;
+		VectorFloat3 [] vertices;
+		VectorFloat2 [] textureCoordinates;
+		short [] indices;
+		VectorFloat3 [] boundaryVertices;
+
+		public ARPlaneGeometryPoker () : base (IntPtr.Zero)
+		{
+		}
+
+		public override nuint VertexCount => 2;
+
+		public override nuint TextureCoordinateCount => 2;
+
+		// There are always 3x more 'TriangleIndices' than 'TriangleCount' since 'TriangleIndices' represents Triangles (set of three indices).
+		// So 2 'TriangleCount' = 6 'TriangleIndices'.
+		public override nuint TriangleCount => 2;
+
+		public override nuint BoundaryVertexCount => 2;
+
+		public override IntPtr GetRawVertices ()
+		{
+			vertices = new VectorFloat3 [] { new VectorFloat3 (1, 2, 3), new VectorFloat3 (4, 5, 6) };
+			if (!verticesArrayHandle.IsAllocated)
+				verticesArrayHandle = GCHandle.Alloc (vertices, GCHandleType.Pinned);
+			return verticesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		public override IntPtr GetRawTextureCoordinates ()
+		{
+			textureCoordinates = new VectorFloat2 [] { new VectorFloat2 (1, 2), new VectorFloat2 (3, 4) };
+			if (!textureCoordinatesArrayHandle.IsAllocated)
+				textureCoordinatesArrayHandle = GCHandle.Alloc (textureCoordinates, GCHandleType.Pinned);
+			return textureCoordinatesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		public override IntPtr GetRawTriangleIndices ()
+		{
+			// Two triangles (set of 3 indices)
+			indices = new short [] { 1, 2, 3, 4, 5, 6 };
+			if (!indicesArrayHandle.IsAllocated)
+				indicesArrayHandle = GCHandle.Alloc (indices, GCHandleType.Pinned);
+			return indicesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		public override IntPtr GetRawBoundaryVertices ()
+		{
+			boundaryVertices = new VectorFloat3 [] { new VectorFloat3 (1, 2, 3), new VectorFloat3 (4, 5, 6) };
+			if (!boundaryVerticesArrayHandle.IsAllocated)
+				boundaryVerticesArrayHandle = GCHandle.Alloc (boundaryVertices, GCHandleType.Pinned);
+			return boundaryVerticesArrayHandle.AddrOfPinnedObject ();
+		}
+
+		protected override void Dispose (bool disposing)
+		{
+			base.Dispose (disposing);
+			if (verticesArrayHandle.IsAllocated)
+				verticesArrayHandle.Free ();
+			if (textureCoordinatesArrayHandle.IsAllocated)
+				textureCoordinatesArrayHandle.Free ();
+			if (indicesArrayHandle.IsAllocated)
+				indicesArrayHandle.Free ();
+			if (boundaryVerticesArrayHandle.IsAllocated)
+				boundaryVerticesArrayHandle.Free ();
+		}
+	}
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class ARPlaneGeometryTest {
+
+		[SetUp]
+		public void Setup ()
+		{
+			TestRuntime.AssertXcodeVersion (9, 0);
+		}
+
+		[Test]
+		public void VerticesTest ()
+		{
+			var face = new ARPlaneGeometryPoker ();
+			var vertices = face.GetVertices ();
+			Assert.AreEqual (new VectorFloat3 (1, 2, 3), vertices [0]);
+			Assert.AreEqual (new VectorFloat3 (4, 5, 6), vertices [1]);
```

Comment:
AFAIK we have as good practice to add a string if we have more than one assert in a test.

Reply:
Oops, I missed that good practice. Minor comment though no (not sure it's blocking this PR)?

Reply:
This wasn't mentioned when the first (similar) tests were added: https://github.com/xamarin/xamarin-macios/pull/3090